src/main/java/uk/gov/hmcts/darts/arm/service/impl/DetsToArmBatchPushProcessorImpl.java

```diff
@@ -0,0 +1,376 @@
+        objectStateRecord.setFlagFileTransfToarml(true);
+        objectStateRecord.setDateFileTransfToarml(currentTimeHelper.currentOffsetDateTime());
+        objectStateRecord.setMd5FileTransfArml(detsExternalObjectDirectory.getChecksum());
+        setFileSize(detsExternalObjectDirectory, objectStateRecord);
+        objectStateRecordRepository.save(objectStateRecord);
+
+    }
+
+    private void copyMetadataToArm(File manifestFile) {
+        try {
+            BinaryData metadataFileBinary = fileOperationService.convertFileToBinaryData(manifestFile.getAbsolutePath());
+            armDataManagementApi.saveBlobDataToArm(manifestFile.getName(), metadataFileBinary);
+        } catch (BlobStorageException e) {
+            if (e.getStatusCode() == BLOB_ALREADY_EXISTS_STATUS_CODE) {
+                log.info("Metadata BLOB already exists", e);
+            } else {
+                log.error("Failed to move BLOB metadata for file {}", manifestFile.getAbsolutePath(), e);
+                throw e;
+            }
+        } catch (Exception e) {
+            log.error("Unable to move BLOB metadata for file {}", manifestFile.getAbsolutePath(), e);
+            throw e;
+        }
+    }
+
+    private void setFileSize(ExternalObjectDirectoryEntity detsExternalObjectDirectory, ObjectStateRecordEntity objectStateRecord) {
+        Long fileSize = getFileSize(detsExternalObjectDirectory);
+        if (nonNull(fileSize)) {
+            objectStateRecord.setFileSizeBytesArml(fileSize);
+        }
+    }
+
+    private Long getFileSize(ExternalObjectDirectoryEntity detsExternalObjectDirectory) {
+        Long fileSize = null;
+        if (nonNull(detsExternalObjectDirectory.getMedia())) {
+            fileSize = mediaRepository.findById(detsExternalObjectDirectory.getMedia().getId()).map(
+                media -> media.getFileSize()).orElse(null);
+        } else if (nonNull(detsExternalObjectDirectory.getAnnotationDocumentEntity())) {
+            fileSize = annotationDocumentRepository.findById(detsExternalObjectDirectory.getAnnotationDocumentEntity().getId()).map(
+                annotationDocument -> Long.valueOf(annotationDocument.getFileSize())).orElse(null);
+        } else if (nonNull(detsExternalObjectDirectory.getCaseDocument())) {
+            fileSize = caseDocumentRepository.findById(detsExternalObjectDirectory.getCaseDocument().getId()).map(
+                caseDocument -> Long.valueOf(caseDocument.getFileSize())).orElse(null);
+        } else if (nonNull(detsExternalObjectDirectory.getTranscriptionDocumentEntity())) {
+            fileSize = transcriptionDocumentRepository.findById(detsExternalObjectDirectory.getTranscriptionDocumentEntity().getId()).map(
+                transcriptionDocument -> Long.valueOf(transcriptionDocument.getFileSize())).orElse(null);
+        }
+        return fileSize;
+    }
```

Comment:
this could live in the EodService, would reduce the complexity of this already complex class